x-pack/legacy/plugins/monitoring/server/es_client/parse_elasticsearch_config.ts

```diff
@@ -0,0 +1,108 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { readFileSync } from 'fs';
+import { readPkcs12Truststore, readPkcs12Keystore } from '../../../../../../src/core/utils';
+
+const KEY = 'xpack.monitoring.elasticsearch';
+
+/*
+ * Parse a config object's Elasticsearch configuration, reading any
+ * certificates/keys from the filesystem
+ */
+export const parseElasticsearchConfig = (config: any) => {
+  const es = config.get(KEY);
+
+  const errorPrefix = `[config validation of [${KEY}].ssl]`;
+  if (es.ssl?.key && es.ssl?.keystore?.path) {
+    throw new Error(`${errorPrefix}: cannot use [key] when [keystore.path] is specified`);
+  }
+  if (es.ssl?.certificate && es.ssl?.keystore?.path) {
+    throw new Error(`${errorPrefix}: cannot use [certificate] when [keystore.path] is specified`);
+  }
+
+  const { alwaysPresentCertificate, verificationMode } = es.ssl;
+  const { key, keyPassphrase, certificate, certificateAuthorities } = readKeyAndCerts(es);
+
+  return {
+    ...es,
+    ssl: {
+      alwaysPresentCertificate,
+      key,
+      keyPassphrase,
+      certificate,
+      certificateAuthorities,
+      verificationMode,
+    },
+  };
+};
+
+const readKeyAndCerts = (rawConfig: any) => {
```

Comment:
Could we maybe export https://github.com/elastic/kibana/blob/80b6dd8e15d0a195a6a84c2f2313902f821d022f/src/core/server/elasticsearch/elasticsearch_config.ts#L269 and use it here as well?

Reply:
As mentioned above, I wanted to avoid a core API change for this bug fix. But this code won't be necessary after the Monitoring plugin moves to the Kibana Platform! I'll add a comment noting that.